migration-engine/ARCHITECTURE.md

```diff
@@ -372,3 +372,34 @@ these issues in Migrate.
       merge conflict in the schema, which should prompt questions (team members
       disagreeing on the datamodel)
 
+### I want to customize a many-to-many relation table (e.g. to add a primary key)
+
+Unfortunately the schema of these tables is a very deep assumption in all of
+migrate, introspection and the query engine, so it's not possible. The long
+term solution is that we want to replace the current system with more explicit
+many-to-many relations where the join table can be specified.
+
+Currently, the only solution would be to stop using implicit many-to-many
+relations (list relation fields on both sides) and use an explicit join table with two
+inline relations:
```

Comment:
Add a note that this will have pretty negative effects on the Prisma Client API and maybe a link to an issue that suggest to fix that.

Reply:
done (last paragraph)